Sources/PostgresNIO/New/PSQLCodable.swift

```diff
@@ -33,6 +34,23 @@ protocol PSQLDecodable {
     ///              to use when decoding json and metadata to create better errors.
     /// - Returns: A decoded object
     static func decode(from byteBuffer: inout ByteBuffer, type: PSQLDataType, format: PSQLFormat, context: PSQLDecodingContext) throws -> Self
+
+    /// Decode an entity from the `byteBuffer` in postgres wire format.
+    /// This method has a default implementation and may be overriden
+    /// only for special cases, like `Optional`s.
+    static func decodeRaw(from byteBuffer: inout ByteBuffer?, type: PSQLDataType, format: PSQLFormat, context: PSQLDecodingContext) throws -> Self
+}
+
+extension PSQLDecodable {
+    @inlinable
+    public static func decodeRaw(from byteBuffer: inout ByteBuffer?, type: PSQLDataType, format: PSQLFormat, context: PSQLDecodingContext) throws -> Self {
+        switch byteBuffer {
+        case .some(var buffer):
+            return try self.decode(from: &buffer, type: type, format: format, context: context)
+        case .none:
+            throw PSQLCastingError.Code.missingData
+        }
```

Comment:
```suggestion
        guard let buffer = byteBuffer else {
            throw PSQLCastingError.Code.missingData
        }
        return try self.decode(from: &buffer, type: type, format: format, context: context)
```
IMO the `guard` makes the intent clearer when reading the code, I tend to reserve switching over options for more complex scenarios. Not a big thing, just a style nit.

Reply:
fixed